src/librados/IoCtxImpl.cc

```diff
@@ -799,16 +797,12 @@ int librados::IoCtxImpl::aio_operate_repair_read(const object_t& oid,
   int op = o->ops[0].op.op;
   ldout(client->cct, 10) << ceph_osd_op_name(op) << " oid=" << oid << " nspace=" << oloc.nspace << dendl;
   // Prepend assert_interval op
-  OSDOp tmp;
-  tmp.op.op = CEPH_OSD_OP_ASSERT_INTERVAL;
-  tmp.op.assert_interval.epoch = e;
-  o->ops.insert(o->ops.begin(), tmp);
-  int size = o->ops.size();
-  o->out_bl.resize(size);
-  o->out_handler.resize(size);
-  o->out_rval.resize(size);
-  o->out_bl[1] = pbl;
-  o->ops[1].op.flags = op_flags;
+  o->assert_interval(e);
```

Comment:
@dzafman i changed the way how you prepend the op, because i think this is less hackery.